### DISCUSSION TITLE
No ERP on btHingeConstraint

### DISCUSSION THREAD

**Discussion by stolk** [bulletphysics/bullet3#3470] [General]
You can only set CFM, not ERP on a hinge.
If you do so, the code asserts.
For the stop, both ERP and CFM can be set.

From the code:

```
            switch(num)
            {
                    case BT_CONSTRAINT_STOP_ERP :
                            m_stopERP = value;
                            m_flags |= BT_HINGE_FLAGS_ERP_STOP;
                            break;
                    case BT_CONSTRAINT_STOP_CFM :
                            m_stopCFM = value;
                            m_flags |= BT_HINGE_FLAGS_CFM_STOP;
                            break;
                    case BT_CONSTRAINT_CFM :
                            m_normalCFM = value;
                            m_flags |= BT_HINGE_FLAGS_CFM_NORM;
                            break;
                    default :
                            btAssertConstrParams(0);
            }
```

**Comment by scpeters** [comment]
We added this to our fork of ODE in Gazebo ([pull request 448](https://bitbucket.org/osrf/gazebo/pull-request/448)).

**Comment by stolk** [comment]
@scpeters I don't understand, this is a Bullet issue, not an OpenDE issue. The pull request pertains to OpenDE not Bullet.

**Comment by scpeters** [comment]
I meant it as a +1, since ODE and Bullet have very similar architectures. We noticed it missing from ODE, so we added it in our fork. So +1 for adding it to bullet as well.

Sorry for being so cryptic.
